tracer/src/Datadog.Trace/Agent/AgentWriter.cs

```diff
@@ -339,7 +339,7 @@ SpanBuffer SwapBuffers()
             }
 
             // Add the current keep rate to the root span
-            var rootSpan = trace.Array[trace.Offset].Context.TraceContext?.RootSpan;
+            var rootSpan = trace.Array[trace.Offset].InternalContext.TraceContext?.RootSpan;
```

Comment:
Is this safe? I could well be wrong, but if we we have mismatched versions, we still have 2 `AgentWriters`, right? 

EDIT: Actually, I think I'm wrong. Struggling to get my head around this 😵

Reply:
Sorry about the confusion, I'm trying to address that in the PR follow-up. If we ever have `Span` objects, then accessing their properties is going to be safe. The only question remaining is "What is the runtime type for Scope.Span?" and we just have to account for it when it is `Datadog.Trace.Span` and when it is not